src/web3/KeychainSigner.ts

```diff
@@ -235,41 +55,15 @@ export class KeychainSigner implements Signer {
   }
 
   async unlock(passphrase: string, duration: number): Promise<boolean> {
-    const privateKey = await getStoredPrivateKey(this.account, passphrase)
-    if (!privateKey) {
-      return false
-    }
-
-    this.unlockedLocalSigner = new LocalSigner(privateKey)
-    this.unlockTime = Date.now()
-    this.unlockDuration = duration
-    return true
-  }
-
-  isUnlocked(): boolean {
-    if (this.unlockDuration === undefined || this.unlockTime === undefined) {
+    const unlocked = await this.lock.unlock(this.account.address, passphrase, duration)
+    if (!unlocked) {
       return false
     }
-
-    if (this.unlockDuration === 0) {
-      return true
-    }
-
-    return this.unlockTime + this.unlockDuration * 1000 > Date.now()
-  }
-
-  /**
-   * Updates the passphrase of an account
-   * @param oldPassphrase - the passphrase currently associated with the account
-   * @param newPassphrase - the new passphrase to use with the account
-   * @returns whether the update was successful
-   */
-  async updatePassphrase(oldPassphrase: string, newPassphrase: string) {
-    const privateKey = await getStoredPrivateKey(this.account, oldPassphrase)
+    const privateKey = await getStoredPrivateKey(this.account, passphrase)
```

Comment:
code golf:
```ts
if (!(unlocked && privateKey)) {
  return false
}
```

Reply:
not quite the same i think. If the account is locked, then i dont want to call the async `getStoredPrivateKey`